utils/bigquery.js

```diff
@@ -62,23 +62,36 @@ async function getParticipantsForNotificationsBQ({
   const bqFieldArray = fieldsToFetch
     .map(convertToBigqueryKey)
     .map((field) => `${field} AS ${field.replace(/\./g, "_DOT_")}`);
-  const queryStr = `SELECT ${bqFieldArray.length === 0 ? "*" : bqFieldArray.join(", ")} 
-    FROM \`Connect.participants\` 
-    WHERE ${bqConditionArray.join(" AND ")} 
-    AND query.notificationSpecIdsUsed.${bqNotificationSpecId} IS NOT true 
-    LIMIT ${limit} OFFSET ${offset}`;
+  const queryStrCommon = `SELECT ${
+    bqFieldArray.length === 0 ? "*" : bqFieldArray.join(", ")
+  } FROM \`Connect.participants\` WHERE ${bqConditionArray.join(" AND ")}`;
 
   try {
+    const queryStr = `${queryStrCommon} AND query.notificationSpecIdsUsed.${bqNotificationSpecId} IS NOT true LIMIT ${limit} OFFSET ${offset}`;
     const [rows] = await bigquery.query(queryStr);
     if (rows.length === 0) return result;
 
     result.hasNext = rows.length === limit;
     result.fetchedDataArray = rows.map((data) => convertToFirestoreData(data));
```

Comment:
Maybe this can be turned into a function to avoid code duplication? 71-75, 81-85

Reply:
I thought about it. 
Current the 4-line code has 2 copies (total 8 lines without empty lines). 
If I add a new function to remove 4 lines of duplicate, the newly created function will be 8 lines, plus 2 lines of function calls. There'll be 10 lines of code after creating the function.
So, I'd defer the new function to a later time when the 4-line code is used more frequently.